src/py/flwr_example/quickstart_pytorch/mnist.py

```diff
@@ -0,0 +1,434 @@
+        
+    client_id: int 
+        Unique integer used for selecting a specif partition. 
+        
+    number_of_clients: int 
+        Total number of clients launched during traning. This value dictates the number of partitions to be created.
+        
+
+    Returns
+    -------
+    data_loader: torch.utils.data.Dataset
+        DataLoader for specific client_id considering number_of_clients partitions.  
+
+    """
+
+    # Set the seed so we are sure to generate the same global batches
+    # indices across all clients
+    np.random.seed(123)
+
+    # Get the data corresponding to this client
+    dataset_size = len(dataset)
+    nb_samples_per_clients = dataset_size // number_of_clients
+    dataset_indices = list(range(dataset_size))
+    np.random.shuffle(dataset_indices)
+
+    # Get starting and ending indices w.r.t CLIENT_ID
+    start_ind = client_id * nb_samples_per_clients
+    end_ind = start_ind + nb_samples_per_clients
+    data_sampler = SubsetRandomSampler(dataset_indices[start_ind:end_ind])
+    data_loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=False, sampler=data_sampler
+    )
+    return data_loader
+
+
+def load_data(
+    data_root: str,
+    train_batch_size: int,
+    test_batch_size: int,
+    cid: int,
+    nb_clients: int,
+) -> Tuple[DataLoader, DataLoader]:
+    """Helper function that loads both training and test datasets for MNIST.
+
+    Parameters
+    ----------
+    data_root: str 
+        Directory where MNIST dataset will be stored.
+        
+    train_batch_size: int 
+        Mini-batch size for training set.
+        
+    test_batch_size: int 
+        Mini-batch size for test set.
+        
+    cid: int 
+        Client ID used to select a specific partition.
+        
+    nb_clients: int 
+        Total number of clients launched during traning. This value dictates the number of unique to be created.
+        
+
+    Returns
+    -------
+    (train_loader, test_loader): Tuple[DataLoader, DataLoader]
+        Tuple contaning DataLoaders for training and test sets.
+
+    """
+
+    transform = transforms.Compose(
+        [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]
+    )
+
+    train_dataset = datasets.MNIST(
+        data_root, train=True, download=True, transform=transform
+    )
+    test_dataset = datasets.MNIST(data_root, train=False, transform=transform)
+
+    # Create partitioned datasets based on the total number of clients and client_id
+    train_loader = dataset_partitioner(
+        dataset=train_dataset,
+        batch_size=train_batch_size,
+        client_id=cid,
+        number_of_clients=nb_clients,
+    )
+
+    test_loader = dataset_partitioner(
+        dataset=test_dataset,
+        batch_size=test_batch_size,
+        client_id=cid,
+        number_of_clients=nb_clients,
+    )
+
+    return (train_loader, test_loader)
+
+
+class MNISTNet(nn.Module):
```

Comment:
Can this be `class MNISTNet(nn.ModuleList)` to align with the type of the first parameter for `train(...)`?

Reply:
ModuleList was not necessary so I changed the other ones in train and test in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .